homeassistant/components/awair/sensor.py

```diff
@@ -1,248 +1,250 @@
-        _LOGGER.error("Unexpected Awair API error: %s", error)
-
-    raise PlatformNotReady
+    """Import Awair configuration from YAML."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: ConfigType,
+    async_add_entities: Callable[[List[Entity], bool], None],
+):
+    """Set up Awair sensor entity based on a config entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    sensors = []
+
+    data: List[AwairResult] = coordinator.data.values()
+    for result in data:
+        if result.air_data:
+            sensors.append(AwairSensor(API_SCORE, result.device, coordinator))
+            device_sensors = result.air_data.sensors.keys()
+            for sensor in device_sensors:
+                if sensor in SENSOR_TYPES:
+                    sensors.append(AwairSensor(sensor, result.device, coordinator))
+
+            # The "DUST" sensor for Awair is a combo pm2.5/pm10 sensor only
+            # present on first-gen devices in lieu of separate pm2.5/pm10 sensors.
+            # We handle that by creating fake pm2.5/pm10 sensors that will always
+            # report identical values, and we let users decide how they want to use
+            # that data - because we can't really tell what kind of particles the
+            # "DUST" sensor actually detected. However, it's still useful data.
+            if API_DUST in device_sensors:
+                for alias_kind in DUST_ALIASES:
+                    sensors.append(AwairSensor(alias_kind, result.device, coordinator))
+
+    async_add_entities(sensors)
 
 
 class AwairSensor(Entity):
-    """Implementation of an Awair device."""
-
-    def __init__(self, data, device, sensor_type, throttle):
-        """Initialize the sensor."""
-        self._uuid = device[CONF_UUID]
-        self._device_class = SENSOR_TYPES[sensor_type]["device_class"]
-        self._name = f"Awair {self._device_class}"
-        unit = SENSOR_TYPES[sensor_type]["unit_of_measurement"]
-        self._unit_of_measurement = unit
-        self._data = data
-        self._type = sensor_type
-        self._throttle = throttle
+    """Defines an Awair sensor entity."""
+
+    def __init__(
+        self, kind: str, device: AwairDevice, coordinator: AwairDataUpdateCoordinator,
+    ) -> None:
+        """Set up an individual AwairSensor."""
+        LOGGER.debug("Setting up %s: %s", kind, device.uuid)
+        self.__kind = kind
+        self.__device = device
+        self.__coordinator = coordinator
+
+        entity_id_name = SENSOR_TYPES[self.__kind][ATTR_DEVICE_CLASS]
+        if self.__device.name:
+            entity_id_name = f"{self.__device.name} {entity_id_name}"
+
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, entity_id_name, hass=self.__coordinator.hass
+        )
```

Comment:
Please don't set the entity_id explicitly, but let the name property be a template for the entity_id. The user can change those at will since we set a unique_id.
```suggestion
        self._kind = kind
        self._device = device
        self._coordinator = coordinator
```